src/app/shared/ag-grid-renderers/clickable-button-renderer/clickable-button-renderer.component.ts

```diff
@@ -12,28 +12,27 @@ export class ClickableButtonRendererComponent implements ICellRendererAngularCom
     label: string;
     isIcon = false;
     icon: string;
+    color: string;
     private params: any;
 
-    agInit(params: any): void {
-        this.params = params;
+    setRendererParams(params) {
         this.label = params.label;
-        if (params.useIcon) {
-            this.isIcon = true;
+        this.isIcon = params.useIcon;
+        if (this.isIcon) {
             this.icon = params.icon;
+            this.color = params.color;
         }
     }
 
+    agInit(params: any): void {
+        this.params = params;
+        this.setRendererParams(params);
+    }
+
     refresh(params: ICellRendererParams): boolean {
         const cellRendererParams = params.colDef.cellRendererParams;
-        let cellParams;
-        if (_.isFunction(cellRendererParams)) {
-            cellParams = cellRendererParams(params);
-        } else {
-            cellParams = cellRendererParams;
-        }
-        this.label = cellParams.label;
-        this.isIcon = cellParams.useIcon;
-        this.icon = cellParams.icon;
+        const cellParams = _.isFunction(cellRendererParams) ? cellRendererParams(params): cellRendererParams;
```

Comment:
Check if this is the correct way to use cellRendererParams.

Reply:
We can provide a function for `cellRendererParams` while defining the column definitions of the ag-grid, it is better to do in this way, if we need to update the renderer params or we need to use the grid params to render a cell (using that renderer). We can also check if we can do the same in other places in the application.